src/ol/source/DataTile.js

```diff
@@ -172,6 +198,61 @@ class DataTileSource extends TileSource {
    * @return {!DataTile} Tile.
    */
   getTile(z, x, y, pixelRatio, projection) {
+    const sourceProjection = this.getProjection();
+    if (
+      sourceProjection &&
+      projection &&
+      !equivalent(sourceProjection, projection)
+    ) {
+      const cache = this.getTileCacheForProjection(projection);
+      let tile;
+      const tileCoordKey = getKeyZXY(z, x, y);
+      if (cache.containsKey(tileCoordKey)) {
+        tile = cache.get(tileCoordKey);
+      }
+      const key = this.getKey();
+      if (tile && tile.key == key) {
+        return tile;
+      } else {
```

Comment:
Minor, but we can have less indentation on the happy path if we avoid these return-else.  I've pushed a commit that reworks these and will add a linter rule separately.